### PR TITLE
MGMT-12743: adds enhancement proposal for shorter image URLs

### PR DESCRIPTION
**Pull request by mhrivnak**
## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [x] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: docs/enhancements/short-url-enhancement.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Today's unsigned URLs look like this:
+
+```
+http://api.openshift.com/api/assisted-images/images/881dd303-940e-4c1a-a844-dc3c64da0e95?arch=x86_64&type=full-iso&version=4.11
+```
+
+That should change to:
+
+```
+http://api.openshift.com/api/assisted-images/images/881dd303-940e-4c1a-a844-dc3c64da0e95/4.11/x86_64/full.iso
```
Comment (omertuc) [inline docs/enhancements/short-url-enhancement.md:82]: Shouldn't this be:
```suggestion
http://api.openshift.com/api/assisted-images/bytoken/images/<TOKEN>/4.11/x86_64/full.iso
```

Comment (carbonin) [reply docs/enhancements/short-url-enhancement.md:82]: In this case I'm not sure `bytoken` makes sense when we have no token since this is unsigned.

Comment (omertuc) [reply docs/enhancements/short-url-enhancement.md:82]: sorry ignore my comment it doesn't make sense

**Comment by carbonin** [comment]
I'm okay with this.

@mhrivnak did you want to add something about the ACM vs SaaS tokens or leave that to sort out during implementation?

File: docs/enhancements/short-url-enhancement.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+That should change to:
+
+```
+http://api.openshift.com/api/assisted-images/byid/881dd303-940e-4c1a-a844-dc3c64da0e95/4.11/x86_64/full.iso
+```
+
+#### Compatibility
+
+The previous URL layouts should continue to work. But the default URLs returned
```
Comment (nmagnezi) [inline docs/enhancements/short-url-enhancement.md:87]: Do we keep the older format for backward compatibility? 
If yes, can we somehow mark (or just document) that the old format is deprecated?

**Review by nmagnezi — COMMENTED** [review]
Looks good.
I have one question that I posted as a comment.

File: docs/enhancements/short-url-enhancement.md (hunk 2 above)
Comment (carbonin) [reply docs/enhancements/short-url-enhancement.md:87]: I don't think we can mark it as deprecated, but at the same time no one should be relying on the format.

The URL is always given to the user in full so they shouldn't care about how it's formatted. I see it as the same as when we moved from s3 to the image service, the URL will be different, but it should be transparent to the user.

Comment (mhrivnak) [reply docs/enhancements/short-url-enhancement.md:87]: I think we need compatibility just long enough to make sure that any URL handed out recently continue to work after the service gets upgraded.

**Comment by mhrivnak** [comment]
> @mhrivnak did you want to add something about the ACM vs SaaS tokens or leave that to sort out during implementation?

I'd like to nail down how ACM handles tokens before accepting this proposal.

**Review by carbonin — APPROVED** [review]

**Pull request merged**